functions/src/index.ts

```diff
@@ -103,41 +115,153 @@ export const notifyCharityAboutDonation = functions.firestore
   });
 
 /**
- * Is triggereed when a document in the "boxes" collection is updated.
- * Notifies the charity when the quantity of a certain box type at the canteen is below 10.
+ * Function triggered when there is an update in the "entityPairs" collection in foodboxes list.
+ * It notifies a charity about the lack of boxes at a canteen.
  *
- * @param change - The updated box document snapshot.
- * @param context - The context object containing metadata about the update event.
- * @returns A promise that resolves when the notification is sent, or null if conditions are not met.
+ * @param change - The change event that triggered the function.
+ * @param context - The context of the function execution.
+ * @returns A promise that resolves when the notification is sent successfully.
  */
 export const notifyCharityAboutLackOfBoxesAtCanteen = functions.firestore
-  .document("boxes/{id}")
+  .document("entityPairs/{id}")
   .onUpdate((change, context) => {
-    const data = change.after.data();
-    const numberOfBoxes = data.quantityAtCanteen;
-    const charityId = data.charityId;
-    const boxType = data.boxType;
-
-    if (numberOfBoxes < 10) {
-      return admin.firestore().collection("fCMTokens").doc(charityId).get()
-        .then((tokenDoc) => {
-          if (tokenDoc.exists) {
-            const fcmToken = tokenDoc.data()?.token;
-
-            const message = {
-              notification: {
-                title: "Jídelně docházejí krabičky",
-                body: `Objednejte prosím svoz krabiček typu "${boxType}" do jídelny`,
-              },
-              token: fcmToken,
-            };
-
-            return admin.messaging().send(message);
+    const newBoxes = change.after.data().foodboxes;
+    const oldBoxes = change.before.data().foodboxes;
+
+    const differenceMap: { [foodBoxId: string]: number } = {};
+    newBoxes.forEach((newBox: FoodBox) => {
+      const matchingOldBox = oldBoxes.find(
+        (oldBox: FoodBox) => oldBox.foodBoxId === newBox.foodBoxId
+      );
+      if (matchingOldBox && newBox.donorCount < matchingOldBox.donorCount) {
+        const difference = newBox.donorCount - matchingOldBox.donorCount;
+        differenceMap[newBox.foodBoxId] = difference;
+      }
+    });
```

Comment:
Q: Why this diff is needed if the difference (`newBox.donorCount - matchingOldBox.donorCount`) is not used for further logic?

Reply:
Good catch, it's used only for logging purposes. I'd like to leave it this way for the first few weeks to see if it behave correctly and to have easier debugging in case of some problems.